src/Mailer.hs

```diff
@@ -113,4 +113,6 @@ mailer settings = reactivelyRunAction
         pending = condenseConsecutive $ usPendingEmails userState
         emails = generateEmail settings uuid userState <$> pending
       in
-        sendEmails settings emails >> return (Emailed <$> pending)
+        case emails of
+          [] -> return []
```

Comment:
How does this case not work out on empty list?

Reply:
The expression is producing an `IO` action that sends emails and then returns events to record which emails it sent. So, this case performs no work, just returns an empty list of events in a side-effect-free `IO` context.

Reply:
Put another way, if we did send emails with an empty list, it would still do all the connecting. Perhaps it would be better to subsume this case into `sendEmails`?

Reply:
That sounds sensible, from out here it was hard to see that it was going to do connecting.